data_structures/hashing/bloom_filter.py

```diff
@@ -0,0 +1,107 @@
+"""
+See https://en.wikipedia.org/wiki/Bloom_filter
+
+The use of this data structure is to test membership in a set.
+Compared to Python's built-in set() it is more space-efficient.
+In the following example, only 8 bits of memory will be used:
+>>> bloom = Bloom(size=8)
+>>> "Titanic" in bloom
+False
+
+Initially the filter contains all zeros:
+>>> bloom.bitstring
+'00000000'
+
+When an element is added, two bits are set to 1
+since there are 2 hash functions in this implementation:
+>>> bloom.add("Titanic")
+>>> bloom.bitstring
+'01100000'
+>>> "Titanic" in bloom
+True
+
+However, sometimes only one bit is added
+because both hash functions return the same value
+>>> bloom.add("Avatar")
+>>> bloom.format_hash("Avatar")
+'00000100'
+>>> bloom.bitstring
+'01100100'
+
+Not added elements should return False ...
+>>> "The Goodfather" in bloom
+False
+>>> bloom.format_hash("The Goodfather")
+'00011000'
+>>> "Interstellar" in bloom
+False
+>>> bloom.format_hash("Interstellar")
+'00000011'
+>>> "Parasite" in bloom
+False
+>>> bloom.format_hash("Parasite")
+'00010010'
+>>> "Pulp Fiction" in bloom
+False
+>>> bloom.format_hash("Pulp Fiction")
+'10000100'
+
+but sometimes there are false positives:
+>>> "Ratatouille" in bloom
+True
+>>> bloom.format_hash("Ratatouille")
+'01100000'
+
+The probability increases with the number of added elements
```

Comment:
Does the probability decrease if the size of the bitarray is increased?

Reply:
Yes. In wikipedia `m` stands for the size of the filter in bits:
![2023-04-08-170542_](https://user-images.githubusercontent.com/10489506/230728556-9891415e-8982-472a-b86a-b4039853229e.png)

To test that we would need to create another `Bloom` object

Reply:
If you  want to go deeper, here is a working test that I removed

https://github.com/isidroas/Python/blob/8b1bec0dc6d5474d749210a70c0221f6527b2258/data_structures/hashing/bloom_filter.py#L104

Reply:
```suggestion
The probability increases with the number of elements added.
The probability decreases with the number of bits in the bitarray.
```